Sources/MongoSwift/BSON/BsonValue.swift

```diff
@@ -357,16 +357,20 @@ extension Double: BsonValue {
 /// The `Int` will be encoded as an Int32 if possible, or an Int64 if necessary.
 extension Int: BsonValue {
 
-    public var bsonType: BsonType { return .int32 }
+    public var bsonType: BsonType { return self.int32Value != nil ? .int32 : .int64 }
```

Comment:
I originally commented that these Int changes did not seem relevant to "check for existing key and overwrite values", but I realize they are appropriate given the Overwritable protocol you introduce later. :+1: